### PR TITLE
excluded EntityAlreadyExists exceptions from requeuing

### PR DESCRIPTION
**Pull request by a-sayyed**
Added `EntityAlreadyExistsException` to the ignored exceptions from the rabbitmq requeuing

### CODE REVIEW

**Comment by laverman** [comment]
@hawkbit-bot please verify

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java [hunk 1]
```diff
@@ -164,7 +165,7 @@ public Message onMessage(final Message message, final String type, final String
             default:
                 logAndThrowMessageError(message, "No handle method was found for the given message type.");
             }
-        } catch (final IllegalArgumentException ex) {
+        } catch (final IllegalArgumentException | EntityAlreadyExistsException ex) {
```
Comment (a-sayyed) [inline hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java]: move the catch to `registerTarget`

File: hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java [hunk 2]
```diff
@@ -418,4 +419,8 @@ private boolean isMultiAssignmentsEnabled() {
                 .runAsSystem(() -> tenantConfigurationManagement.getConfigurationValue(key, valueType).getValue());
     }
 
-}
\ No newline at end of file
+    // for testing
+    public void setControllerManagement(final ControllerManagement controllerManagement) {
```
Comment (a-sayyed) [inline hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java]: reduce visibility

File: hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageHandlerServiceIntegrationTest.java [hunk 3]
```diff
@@ -882,6 +891,25 @@ public void downloadOnlyAssignmentAllowsActionStatusUpdatesWhenTargetReportsFini
         verifyAssignedDsAndInstalledDs(controllerId, distributionSet.getId(), distributionSet.getId());
     }
 
+    @Test
+    @Description("Messages that result into certain exceptions being raised should not be requeued. This message should forwarded to the deadletter queue")
```
Comment (schabdo) [inline hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageHandlerServiceIntegrationTest.java]: > result into certain exceptions

It is only `EntityAlreadyExistsException` so why not mentioning it?

File: hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerServiceIntegrationTest.java [hunk 4]
```diff
@@ -6,13 +6,17 @@
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-v10.html
  */
-package org.eclipse.hawkbit.integration;
+package org.eclipse.hawkbit.amqp;
```
Comment (schabdo) [inline hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerServiceIntegrationTest.java]: Due to the move JavaDoc link of `AbstractAmqpServiceIntegrationTest` is broken

Comment (schabdo) [inline hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerServiceIntegrationTest.java]: Hierarchy looks strange now: 

```
org.eclipse.hawkbit.integration.AbstractAmqpServiceIntegrationTest
=> org.eclipse.hawkbit.integration.AmqpMessageDispatcherServiceIntegrationTest
=> org.eclipse.hawkbit.amqp.AmqpMessageHandlerServiceIntegrationTest
```

File: hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageHandlerServiceIntegrationTest.java (hunk 3 above)
Comment (a-sayyed) [reply hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageHandlerServiceIntegrationTest.java]: I added `IllegalArgumentException` to be tested as well

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Pull request merged**